internal/serve/httphandler/payments_handler.go

```diff
@@ -22,13 +22,19 @@ func (h PaymentsHandler) SubscribeAddress(w http.ResponseWriter, r *http.Request
 	var reqBody PaymentsSubscribeRequest
 	err := httpdecode.DecodeJSON(r, &reqBody)
 	if err != nil {
-		httperror.BadRequest("Invalid request body").Render(w)
+		httperror.BadRequest("Invalid request body", nil).Render(w)
+		return
+	}
+
+	httpErr := ValidateRequestBody(ctx, reqBody)
+	if httpErr != nil {
+		httpErr.Render(w)
 		return
 	}
```

Comment:
💡  I wonder if we could add a method that applies this same logic consistently every time. Something like 

```go
func DecodeJSONAndValidate(r *http.Request, ptr interface{}) httperror.Response {
	err := httpdecode.DecodeJSON(r, ptr)  // ptr is the same as &reqBody
	if err != nil {
		return httperror.BadRequest("Invalid request body", nil)
	}

	return ValidateRequestBody(ctx, reqBody)
}
```

Reply:
Awesome!